lib/core/src/system/web/dom/html/css3d_renderer.rs

```diff
@@ -0,0 +1,269 @@
+    }
+}
+
+// ====================
+// === Css3dRenderer ===
+// ====================
+
+/// A renderer for `Css3dObject`s.
+#[derive(Clone,Debug)]
+pub struct Css3dRenderer {
+    container      : DomContainer,
+    data           : Rc<Css3dRendererData>,
+    logger         : Logger
+}
+
+impl Css3dRenderer {
+    /// Creates a Css3dRenderer inside an element.
+    pub fn from_element<L:Into<Logger>>(logger:L, element:HtmlElement) -> Result<Self> {
+        let logger                     = logger.into();
+        let container                  = DomContainer::from_element(element);
+        let front_dom    : HtmlElement = dyn_into(create_element("div")?)?;
+        let back_dom     : HtmlElement = dyn_into(create_element("div")?)?;
+        let front_camera : HtmlElement = dyn_into(create_element("div")?)?;
+        let back_camera  : HtmlElement = dyn_into(create_element("div")?)?;
+
+        front_dom.set_property_or_panic("position", "absolute");
+        front_dom.set_property_or_panic("top", "0px");
+        front_dom.set_property_or_panic("overflow", "hidden");
+        front_dom.set_property_or_panic("overflow", "hidden");
+        front_dom.set_property_or_panic("width", "100%");
+        front_dom.set_property_or_panic("height", "100%");
+        front_dom.set_property_or_panic("pointer-events", "none");
+        back_dom.set_property_or_panic("position", "absolute");
+        back_dom.set_property_or_panic("top", "0px");
+        back_dom.set_property_or_panic("overflow", "hidden");
+        back_dom.set_property_or_panic("overflow", "hidden");
+        back_dom.set_property_or_panic("width", "100%");
+        back_dom.set_property_or_panic("height", "100%");
+        back_dom.set_property_or_panic("pointer-events", "none");
+        back_dom.set_property_or_panic("z-index", "-1");
+        front_camera.set_property_or_panic("width", "100%");
+        front_camera.set_property_or_panic("height", "100%");
+        front_camera.set_property_or_panic("transform-style", "preserve-3d");
+        back_camera.set_property_or_panic("width", "100%");
+        back_camera.set_property_or_panic("height", "100%");
+        back_camera.set_property_or_panic("transform-style", "preserve-3d");
+
+        container.dom.append_or_panic(&front_dom);
+        container.dom.append_or_panic(&back_dom);
+        front_dom.append_or_panic(&front_camera);
+        back_dom.append_or_panic(&back_camera);
+
+        let data             = Css3dRendererData::new(front_dom, back_dom, front_camera, back_camera);
+        let data             = Rc::new(data);
+        let mut renderer     = Self {container,data,logger};
+
+        renderer.init_listeners();
+        Ok(renderer)
+    }
+
+    /// Creates a Css3dRenderer.
+    pub fn new<L:Into<Logger>>(logger:L, dom_id: &str) -> Result<Self> {
+        Self::from_element(logger,dyn_into(get_element_by_id(dom_id)?)?)
+    }
+
+    pub(crate) fn logger(&self) -> Logger {
+        self.logger.clone()
+    }
```

Comment:
clonning logger is pricy now (as it contains strings). What is it needed for?

Reply:
@wdanilo it was needed in Css3dSystem::new:

```rust
    pub fn new(world:&World) -> Self {
        let scene          = world.scene();
        let css3d_renderer = scene.css3d_renderer();
        let logger         = css3d_renderer.logger().sub("Css3dSystem");
        let display_object = DisplayObjectData::new(&logger);
        Self{display_object,css3d_renderer,logger}
    }
```

but this code was refactored to:

```rust
impl Css3dSystem {
    pub fn new(world:&World) -> Self {
        let scene          = world.scene();
        let css3d_renderer = scene.css3d_renderer();
        css3d_renderer.new_system()
    }
```

and in `Css3dRenderer`:

```rust
impl Css3dRenderer {
    pub(super) fn new_system(&self) -> Css3dSystem {
        let css3d_renderer = self.clone();
        let logger         = self.logger.sub("Css3dSystem");
        let display_object = DisplayObjectData::new(&logger);
        Css3dSystem {display_object,css3d_renderer,logger}
    }
```

Reply:
I made it like this because `Css3dSystem` is tied to `Css3dRenderer`, so it should be an instance created by `Css3dRenderer`.